code/modules/modular_computers/networking/_network.dm

```diff
@@ -172,4 +184,19 @@ GLOBAL_LIST_EMPTY(computer_networks)
 	for(var/id in GLOB.computer_networks)
 		var/datum/computer_network/net = GLOB.computer_networks[id]
 		if(net.router && ARE_Z_CONNECTED(get_z(net.router.holder), get_z(T)))
-			return net
\ No newline at end of file
+			return net
+
+// Takes a list of network_device datums and an executor and only returns devices
+// which pass an access check.
+/datum/computer_network/proc/filter_devices_by_access(var/list/network_devices, var/mob/user)
+	// if administrator, give full access.
+	var/obj/item/card/id/network/id = user.GetIdCard()
+	if(id && istype(id, /obj/item/card/id/network) && access_controller && (id.user_id in access_controller.administrators))
+		return network_devices
+
+	var/list/allowed_devices = list()
+	for(var/datum/extension/network_device/D in network_devices)
+		var/obj/machine = D.holder
```

Comment:
/obj, network_device is not guaranteed to be on a machine.

Reply:
It's not on a machine, that's just an /obj reference. All network_devices are /objs.

Reply:
to clear it up I renamed the var from "machine" to "M" so it's not confusable.